java/parser/Parser.java

```diff
@@ -506,33 +540,51 @@ public Pattern visitPattern(final GraqlParser.PatternContext ctx) {
     }
 
     @Override
-    public Pattern visitPattern_disjunction(final GraqlParser.Pattern_disjunctionContext ctx) {
+    public Disjunction<? extends Pattern> visitPattern_disjunction(final GraqlParser.Pattern_disjunctionContext ctx) {
         final List<Pattern> patterns = ctx.patterns().stream().map(patternsContext -> {
             final List<Pattern> nested = visitPatterns(patternsContext);
             if (nested.size() > 1) return new Conjunction<>(nested);
             else return nested.get(0);
         }).collect(toList());
 
-        // Simplify representation when there is only one alternative
-        if (patterns.size() == 1) {
-            return patterns.iterator().next();
-        }
+        // TODO do we want this?
```

Comment:
discussion point

Reply:
Replace with an assertion that `patterns.size()` is always larger than 1.